docs/DeveloperGuide.md

```diff
@@ -1,38 +1,433 @@
+| `bye`        | `ExitCommand`                      | Exits the program.                                                 |
+| `export`     | `ExportCommand`                    | Creates a portable NUSMod link to create their timetable on NUSMod |
+| `get`        | `GetCommand`                       | Display all details about a module.                                |
+| `help`       | `HelpCommand`                      | Display all possible command words and their usage to user.        |
+| `import`     | `ImportCommand`                    | Import user's timetable from a NUSMod share timetable link.        |
+| `search`     | `SearchModuleCommand`              | Searches similar modules based on code, title, semester or level.  |
+| `semester`   | `SelectSemesterCommand`            | Selects the semester that the user want.                           |
+| `select`     | `SelectSlotCommand`                | Selects the time slot for the different lesson types.              |
+| `view`       | `ViewCommand`                      | Views the user timetable with user's selected modules.             |
+
+<!--
+##### How the feature is implemented
+
+##### Why it is implemented this way.
+
+##### Alternatives considered.
+-->
+
+#### 3.4.1 AddModuleCommand
+
+The <code>AddModuleCommand</code> class extends from the <code>Command</code> class and adds the user input module into
+their timetable.
+
+![AddModuleCommand Class](..\docs\images\addModuleCommandClass.png)
+
+##### How the feature is implemented
+The `AddModuleCommand` class extends the `Command` class.
+The constructor `AddModuleCommand()` parses the user `input` module code `.toUpperCase()` as the format to fetch an
+instance of `module` from its class. Boolean `successful` field is used to flag successfully added modules in comparison
+to instances where it is not possible to add the `module` as it already exists in the `state`'s `selectedModuleList`.
+It overrides the `execute()` method from the `Command` class, and updates `successful` accordingly, which will later be
+passed on to the overridden `getExecutionMessage()` which displays the result of data validation that new `selectedModule`
+added are unique.
+
+##### Why it is implemented this way.
+In order to be able to be able to compare the new instance of `selectedModule` created of the module code the user wants to delete,
+in the constructor against an instance of the module the user has previously added into the `selectedModuleList`, the `equals()`
+method extended from super class `Object` has been overridden to return `true` for instances where `semester` and `module`
+(specifically `moduleCode` attribute from the parent class) are the same, allowing us to validate and add the desired module.
+
+The following sequence diagram shows how the undo operation works:
+
+![AddModuleCommandSequenceDiagram](images/AddModuleCommandSequenceDiagram.png)
```

Comment:
Do consider changing the configuration of your plantUML as the diagrams are off , such as this object box at the bottom of the lifeline.
![image](https://user-images.githubusercontent.com/88614316/198252272-27fa5a3a-57dd-453a-8036-b58964b21a31.png)